src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,116 @@
 public class StringManipulation implements StringManipulationInterface {
+    private String string;
 
-    @Override
     public String getString() {
-        return null;
+        return string;
     }
 
-    @Override
     public void setString(String string) {
+        this.string = string;
     }
 
-    @Override
+    /**
+     * Counts the number of non-empty words in the string.
+     * @return The count of words.
+     */
     public int count() {
-        return 0;
+        String[] words = string.split(" ");
+        int count = 0;
+
+        for (String word : words) {
+            if (!word.isEmpty()) {
+                count++;
+            }
+        }
+
+        return count;
     }
 
+    /**
+     * Removes the character at the nth position in the string.
+     * @param n The position of the character to be removed.
+     * @param maintainSpacing Determines whether to maintain spacing when removing characters.
+     * @return The string with the nth characters removed.
+     */
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
-    }
+        if (n <= 0) {
+            throw new IllegalArgumentException();
+        } else if (n > string.length()) {
+            throw new IndexOutOfBoundsException();
+        }
 
-    @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+
+        for (int i = 0; i < string.length(); i++) {
+            if ((i + 1) % n == 0) {
+                if (maintainSpacing) {
+                    stringBuilder.append(" ");
+                }
+            } else {
+                stringBuilder.append(string.charAt(i));
+            }
+        }
```

Comment:
Your code looks great in removeNthCharacter()! I used basically the same logic for this method:)
The only difference between yours and mine is, I decided to check if(i % n != 0) and in that case, I append the character at the index, and the else if statement checks if (maintspacing), and if so I append a space.